### ISSUE TITLE
Add support for HuggingFace Datasets

### ISSUE THREAD

**Issue by meanderingstream** [elixir-nx/bumblebee#137]
Datasets could be downloaded directly instead of via Bumblebee.  However, the key need is support for Parquet.  It is my understanding that Parquet is going to be the "standard" for HF datasets.  I couldn't find an Elixir library that can explode Parquet.

**Comment by josevalim** [comment]
https://github.com/elixir-nx/explorer :) it can read parquet and other formats and it is quite efficient as its engine is powered by Rust/Polars. :)

**Comment by meanderingstream** [comment]
Explorer isn't reading the HF parquet files that include images.
Example:
Download:
wget https://huggingface.co/datasets/nelorth/oxford-flowers/blob/main/data/train-00000-of-00001-12de94e121bdbead.parquet
Explorer.DataFrame.from_parquet("data/train-00000-of-00001-12de94e121bdbead.parquet")
> {:error, {:polars, "External format error: File out of specification: The file must end with PAR1"}}

**Comment by josevalim** [comment]
Interesting. Can you read it with Python? Which parquet library and code would you use to do that? Then I can start debugging comparing to Python. Thanks!

**Comment by meanderingstream** [comment]
I haven't successfully read the file in.  I'm running into python module problems.  Pandas should support reading parquet files but I need to google some more because it's not finding pyarrow even though I installed it.

from https://huggingface.co/docs/datasets-server/parquet:
import pandas as pd
url = "https://huggingface.co/datasets/duorc/resolve/refs%2Fconvert%2Fparquet/ParaphraseRC/duorc-train.parquet"
pd.read_parquet(url).title.value_counts().head()

I've been asking for help from my HF contact.  He, in turn, asked for some help.   We'll see when I learn more.  The holidays are upon us.  

From our discussions, I think that HF is expecting DL training to use the parquet file as the one file for a dataset source.  Using the Arrow parquet query capability, an image is pulled from the file as needed.  Low memory usage and some speed.

**Comment by meanderingstream** [comment]
I've dug around some more.  There are some reasonable HF datasets that are not provided in parquet.  zip file format was the main alternative.  Therefore, I would judge this request as a medium priority.  There are several datasets that Elixirists can use now.

I had trouble getting pandas to load the HF parquet image files.  For examples of parquet image files, I found at least the following datasets are parquet:
nelorth/oxford-flowers, ceyda/smithsonian_butterflies

I was able to load the parquet files using the Huggingface datasets library using the following guide: https://huggingface.co/docs/datasets/load_hub

conda install -c anaconda python
pip install datasets
pip install datasets[vision]

from datasets import load_dataset_builder
from datasets import load_dataset

ds_builder = load_dataset_builder("nelorth/oxford-flowers")
ds_builder.info.description

dataset = load_dataset("nelorth/oxford-flowers", split="test")
